src/core/or/circuituse.c

```diff
@@ -1984,6 +1984,45 @@ circuit_purpose_is_hidden_service(uint8_t purpose)
    return 0;
 }
 
+/** Retrun true iff the given circuit is an HS client circuit. */
```

Comment:
Hmm, `circuit_purpose_is_hidden_service()` right above these new functions seems very related. How can we reuse these functions there? So that we don't have duplicate purpose-checking code?

Reply:
So this is a "hard" one in terms of API but we could do better I agree.

Basically, we have HS circuit as a "catch all" circuit. Then we have *client* and *service* and then we have *v2* and *v3*.

So the `circuit_purpose_is_hidden_service()` is the catch all basically. I could add a refactor commit to use the new helper function within that big one?

Reply:
Yep that would work. There might need to be some argument interface rearrangemnet but I think it's possible.

Reply:
FIxed in the new PR. Basically, the 3 helpers function now take a purpose and not a circuit so then `circuit_purpose_is_hidden_service()` can use them. It is nicer now as I think it is unified better.